fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java

```diff
@@ -611,8 +611,9 @@ private boolean processModifyColumn(ModifyColumnClause alterClause, OlapTable ol
                 if (!col.equals(modColumn)) {
                     typeChanged = true;
                     // TODO:the case where columnPos is not empty has not been considered
-                    if (columnPos == null && col.getDataType() == PrimitiveType.VARCHAR
-                            && modColumn.getDataType() == PrimitiveType.VARCHAR) {
+                    if (columnPos == null && (col.getDataType() == PrimitiveType.VARCHAR
+                            && modColumn.getDataType() == PrimitiveType.VARCHAR)
+                            || (col.getDataType().isVariantType() && modColumn.getDataType().isVariantType())) {
```

Comment:
Do you mean sc between VariantType and ComplexVariantType?

Reply:
VariantType and VariantType